DigitalLearningSolutions.Data/DataServices/ActivityDataService.cs

```diff
@@ -72,14 +72,21 @@ SELECT ap.ApplicationID
             );
         }
 
-        public DateTime GetStartOfActivityForCentre(int centreId)
+        public DateTime? GetStartOfActivityForCentre(int centreId, int? courseCategoryId = null)
         {
-            return connection.QuerySingleOrDefault<DateTime>(
-                @"SELECT MIN(LogDate)
+            return courseCategoryId == null
+                ? connection.QuerySingleOrDefault<DateTime?>(
+                    @"SELECT MIN(LogDate)
                     FROM tActivityLog
                     WHERE CentreID = @centreId",
-                new { centreId }
-            );
+                    new { centreId }
+                )
+                : connection.QuerySingleOrDefault<DateTime?>(
+                    @"SELECT MIN(LogDate)
+                    FROM tActivityLog
+                    WHERE CentreID = @centreId AND CourseCategoryId = @courseCategoryId",
+                    new { centreId, courseCategoryId }
+                );

```

Comment:
This stuff doesn't need to be 2 separate SQL queries. The pattern we've been using for category ids is `(CourseCategoryID = @courseCategoryId OR @courseCategoryId IS NULL)`

Reply:
That makes sense, thank you!